crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2653,6 +2654,14 @@ impl KnownFunction {
     }
 }
 
+#[salsa::interned]
+pub struct ModuleLiteralType<'db> {
+    /// The name of the module being referenced
+    pub name: ModuleName,
+    /// The definition of the module being referenced
+    pub module: File,
```

Comment:
Nit: I'm slightly leaning towards naming this `file`. It's the file in which the module is defined. Also I was a bit confused what `module_ref.module()` would return: The module of the module?

Reply:
Ah yes, this came from when I was also storing the file containing the reference that was originally resolved to this module, and in that case I was using more descriptive names to distinguish the two files.  But now I'm not, ~so I'll revert back to that name!~

Actually, with the change to store `Module` in here, `module` does (now) seem to be the right name.